frequenz_resampling/_rust_backend.pyi

```diff
@@ -0,0 +1,81 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+__all__ = "Resampler", "ResamplingFunction"
+
+from enum import Enum
+from datetime import datetime, timedelta
+from typing import Optional
+
+class ResamplingFunction(Enum):
+    """
+    The ResamplingFunction enum represents the different resampling functions
+    that can be used to resample a time series.
+    """
+
+    Average = 0
+    """Calculates the average of all samples in the time step (ignoring None values)"""
+    Sum = 1
+    """Calculates the sum of all samples in the time step (ignoring None values)"""
+    Max = 2
+    """Calculates the maximum of all samples in the time step"""
+    Min = 3
+    """Calculates the minimum of all samples in the time step"""
+    Last = 3
+    """Returns the last sample in the time step"""
+    Count = 3
+    """Counts the number of samples in the time step"""
+
+class Resampler:
+    """
+    The Resampler class is used to resample a time series of samples.
+
+    It stores the samples in a buffer and resamples the samples in the buffer when the
+    resample method is called.
+    A resampler can be configured with a resampling function and a resampling interval.
+    """
+
+    def __init__(
+        self,
+        interval: timedelta,
+        resampling_function: ResamplingFunction,
+        max_age_in_intervals: int,
+        start: datetime,
```

Comment:
If this is possible, it would be great. It might not be trivial to know what these arguments are about at the caller site otherwise:

```suggestion
        *,
        max_age_in_intervals: int,
        start: datetime,
```